brave/src/main/java/brave/propagation/B3Propagation.java

```diff
@@ -153,8 +165,20 @@ public FactoryBuilder injectFormats(Span.Kind kind, Format format1, Format forma
     @Override public void inject(TraceContext context, R request) {
       Format[] formats = factory.injectFormats;
       if (request instanceof Request) {
-        Span.Kind kind = ((Request) request).spanKind();
-        formats = factory.kindToInjectFormats.get(kind);
+        switch (((Request) request).spanKind()) {
+          case CLIENT:
+            formats = factory.clientInjectFormats;
```

Comment:
in a future change we can actually return a constant injector, as the formats won't change per call to inject. The advantage would be similar to other places where we have constant vs composite... even though the iteration across an array of size two is fast, it shouldn't be quicker than something with no array. By default, there's only one inject format per kind.

Reply:
oh... I have an idea that's much easier! Add a subtype in request!

Request.Setter extends Propagation.Setter
exposes spanKind()

With this, we can hardwire B3 without needing reflection at runtime except the slow path for things that haven't extended Request.Setter yet.

Reply:
ok I did this :)